icu4c/source/i18n/numparse_compositions.h

```diff
@@ -31,10 +31,18 @@ class U_I18N_API CompositionMatcher : public NumberParseMatcher {
     CompositionMatcher() = default;
 
     // To be overridden by subclasses (used for iteration):
+#ifndef __SUNPRO_CC
     virtual const NumberParseMatcher* const* begin() const = 0;
+#else
+    virtual const NumberParseMatcher* const* begin() const {return nullptr;};
```

Comment:
Why is this one necessary?  There are plenty of other places in ICU where we use pure-virtual classes.